app/authentication/auth.py

```diff
@@ -102,6 +107,10 @@ def requires_auth():
         except TokenDecodeError:
             continue
         except TokenExpiredError:
+            decoded_token = decode_token(auth_token)
```

Comment:
Need to use `decode_token` which does not check the signature/does not throw an exception to extract the `iat` field.

`decode_jwt_token`, used a few lines before, throws an exception.

See https://github.com/alphagov/notifications-python-client/blob/ec19d4dd08e2037df867aa5a7171849183f0b6e3/notifications_python_client/authentication.py#L139-L145